### PR TITLE
TLUNIZIN-853 Tool should be usable by friends

### PR DESCRIPTION
**Pull request by dovek**
Due: 3/24/2016
@dlhaines 
@gsilver

### CODE REVIEW

File: src/main/webapp/assets-lti/js/controllers/controllers.js [hunk 1]
```diff
@@ -3,13 +3,17 @@
...
 /* SINGLE COURSE CONTROLLER */
 canvasSupportApp.controller('courseController', ['Course', 'Courses', 'Sections', 'Friend', 'SectionSet', 'Terms', 'focus', '$scope', '$rootScope', '$filter', function (Course, Courses, Sections, Friend, SectionSet, Terms, focus, $scope, $rootScope, $filter) {
-  
-  $scope.contextCourseId = $rootScope.ltiLaunch.custom_canvas_course_id;
-  $scope.currentUserId = $rootScope.ltiLaunch.custom_canvas_user_login_id;
-  Friend.lookUpCanvasFriend($scope.currentUserId).then(function (resultLookUpCanvasUser) {
-    // Canvas does fuzzy searches on uniqname, need to parse results of search to return the user who is an exact match
-    $scope.canvas_user_id = parseResultLookUpCanvasUser(resultLookUpCanvasUser.data, $scope.currentUserId);
-    //REGEXINFO: canvas.api.get.single.course.regex
+	$scope.userIsFriend=false;
```
Comment (gsilver) [inline src/main/webapp/assets-lti/js/controllers/controllers.js]: Whitespace change: looks like the new block is all indented 2 spaces...

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 2]
```diff
@@ -524,11 +524,12 @@ private void mpathwaysCall(HttpServletRequest request, HttpServletResponse respo
 		}
 		logApiCall(uniqname, originalUrl, request);
 		String mpathwaysTermId = request.getParameter(PARAMETER_TERMID);
-		if(uniqname == null || 
+		if(uniqname == null || uniqname.contains("+") ||
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: Explain why '+' matters.

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 3]
```diff
@@ -524,11 +524,12 @@ private void mpathwaysCall(HttpServletRequest request, HttpServletResponse respo
...
 				mpathwaysTermId == null){
 			response.setStatus(400);
-			wrappedResult = new WAPIResultWrapper(400, "Parameter missing in Instructors request", new JSONObject());
-			M_log.error("Error in mpathwaysCall(), missing parameter in Instuctors request");
+			wrappedResult = new WAPIResultWrapper(400, "Parameter missing in Instructors request OR you are using a friend account",
+					new JSONObject());
+			M_log.error("Error in mpathwaysCall(), missing parameter in Instuctors request OR user is using a friend account");
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: spelling

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 4]
```diff
@@ -524,11 +524,12 @@ private void mpathwaysCall(HttpServletRequest request, HttpServletResponse respo
 		}
 		logApiCall(uniqname, originalUrl, request);
 		String mpathwaysTermId = request.getParameter(PARAMETER_TERMID);
-		if(uniqname == null || 
+		if(uniqname == null || uniqname.contains("+") ||
 				mpathwaysTermId == null){
 			response.setStatus(400);
-			wrappedResult = new WAPIResultWrapper(400, "Parameter missing in Instructors request", new JSONObject());
-			M_log.error("Error in mpathwaysCall(), missing parameter in Instuctors request");
+			wrappedResult = new WAPIResultWrapper(400, "Parameter missing in Instructors request OR you are using a friend account",
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: Use same string for wapi wrapper and log message.

File: src/main/webapp/assets-lti/js/utils.js [hunk 5]
```diff
@@ -55,8 +55,13 @@ var filterOutSections = function(sectionData, mPathArray){
   return sectionData;
 };
 
-var parseResultLookUpCanvasUser = function(userArray, uniqname){
-  var thisUser = _.findWhere(userArray, {sis_login_id: uniqname});
+var parseResultLookUpCanvasUser = function(userArray, uniqname, userIsFriend){
+  if(!userIsFriend){
+    var thisUser = _.findWhere(userArray, {sis_login_id: uniqname});
```
Comment (dlhaines) [inline src/main/webapp/assets-lti/js/utils.js]: Not sure that this belongs here but someone reading the code should know how to find out why it is sis_login_id sometimes and sis_user_id other times.

**Pull request merged**